manager/orchestrator/updater.go

```diff
@@ -135,18 +149,56 @@ func (u *Updater) Run(ctx context.Context, cluster *api.Cluster, service *api.Se
 		}()
 	}
 
-	for _, t := range dirtyTasks {
-		// Wait for a worker to pick up the task or abort the update, whichever comes first.
-		select {
-		case <-u.stopChan:
-			break
+	var failedTaskWatch chan events.Event
+
+	if service.Spec.Update == nil || service.Spec.Update.OnFailure == api.UpdateConfig_PAUSE {
```

Comment:
Not sure about this model.

The update workers are processing the update, shouldn't they just send up errors as they encounter them rather than having a different watch?

Pseudo code:
```go
errCh := make(...)
u.worker(ctx, ..., errCh)

for ... {
select {
case stopchan:
case err := <-errCh
case taskQueue <- t:
}
}
```